host/tests/rmfAudio_L3_TestCases/rmfAudioHelperClass.py

```diff
@@ -0,0 +1,174 @@
+# *
+#* ******************************************************************************
+
+import os
+import sys
+import time
+
+dir_path = os.path.dirname(os.path.realpath(__file__))
+sys.path.append(os.path.join(dir_path, "../"))
+
+from raft.framework.plugins.ut_raft import utHelperClass
+from raft.framework.plugins.ut_raft.utPlayer import utPlayer
+from raft.framework.plugins.ut_raft.configRead import ConfigRead
+from raft.framework.plugins.ut_raft.utUserResponse import utUserResponse
+from raft.framework.core.logModule import logModule
+from rmfAudioClasses.rmfAudio import rmfAudioClass
+
+class rmfAudioHelperClass(utHelperClass):
+    """
+    A helper class to support RMF audio capture testing functionalities, including device setup,
+    configuration loading, asset management, and user interactions for manual test validation.
+
+    This class is initialized with test-specific parameters and includes methods to:
+    - Download test artifacts
+    - Run prerequisite and post-requisite commands
+    - Interact with the RMF audio capture device for manual checks
+    - Manage test assets on the Device Under Test (DUT)
+    """
+
+    def __init__(self, testName:str, qcId:str, log:logModule=None ):
+        """
+        Initializes the rmfAudioHelperClass with test-specific details, including test name,
+        configuration paths, and logging setup.
+
+        Args:
+            testName (str): Name of the test.
+            qcId (str): Quality control ID for the test.
+            log (logModule, optional): Logging module instance. Defaults to None.
+        """
+        self.testName  = ""
+        self.testSetupPath = os.path.join(dir_path, "rmfAudio_L3_testSetup.yml")
+        self.moduleName = "rmfaudiocapture"
+        self.rackDevice = "dut"
+
+        super().__init__(testName, qcId, log)
+
+        # Load test setup configuration
+        self.testSetup = ConfigRead(self.testSetupPath, self.moduleName)
+
+        # Open Session for hal test
+        self.hal_session = self.dut.getConsoleSession("ssh_hal_test")
+        self.player_session = self.dut.getConsoleSession("ssh_player")
+
+        cmds = self.testSetup.get("assets").get("device").get(self.testName).get("postcmd")
+        if cmds is not None:
+            for cmd in cmds:
+                self.hal_session.write(cmd)
+
+        # Set up paths and URLs for device test setup
+        deviceTestSetup = self.cpe.get("test")
+        socVendor = self.cpe.get("soc_vendor")
+
+        # Create player Class
+        self.testPlayer = utPlayer(self.player_session, socVendor)
+
+         # Create user response Class
+        self.testUserResponse = utUserResponse()
+
+        # Get path to device profile file
+        self.moduleConfigProfileFile = os.path.join(dir_path, deviceTestSetup.get("profile"))
+        self.targetWorkspace = self.cpe.get("target_directory")
+        self.targetWorkspace = os.path.join(self.targetWorkspace, self.moduleName)
+
```

Comment:
add 
self.streamDownloadURL = deviceTestSetup.get("streams_download_url")
other wise streams will not downloaded

Reply:
updated as suggested